lib/WPA/Andersen.cpp

```diff
@@ -505,7 +505,7 @@ bool Andersen::collapseField(NodeID nodeId)
         if (fieldId != baseId)
         {
             // use the reverse pts of this field node to find all pointers point to it
-            const NodeBS &revPts = getRevPts(fieldId);
+            const NodeBS revPts = getRevPts(fieldId);
```

Comment:
Why remove the reference?

Reply:
The `clearPts` call in the loop (over `revPts`) will invalidate the current iterator into `revPts`, I believe, since `clearPts` will call `clearSingleRevPts`.